pkg/operator/ceph/cluster/mgr/dashboard.go

```diff
@@ -121,27 +121,36 @@ func (c *Cluster) configureDashboardModule(m *mgrConfig) error {
 		return err
 	}
 
-	// server port
-	port := strconv.Itoa(m.DashboardPort)
-	changed, err := client.MgrSetConfig(c.context, c.Namespace, m.DaemonID, c.clusterInfo.CephVersion, "mgr/dashboard/server_port", port, false)
-	if err != nil {
-		return err
-	}
-	hasChanged = hasChanged || changed
-
 	// ssl support
 	var ssl string
 	if c.dashboard.SSL == nil {
 		ssl = ""
 	} else {
 		ssl = strconv.FormatBool(*c.dashboard.SSL)
 	}
-	changed, err = client.MgrSetConfig(c.context, c.Namespace, m.DaemonID, c.clusterInfo.CephVersion, "mgr/dashboard/ssl", ssl, false)
+	changed, err := client.MgrSetConfig(c.context, c.Namespace, m.DaemonID, c.clusterInfo.CephVersion, "mgr/dashboard/ssl", ssl, false)
 	if err != nil {
 		return err
 	}
 	hasChanged = hasChanged || changed
 
+	// server port
+	port := strconv.Itoa(m.DashboardPort)
+	changed, err = client.MgrSetConfig(c.context, c.Namespace, m.DaemonID, c.clusterInfo.CephVersion, "mgr/dashboard/server_port", port, false)
+	if err != nil {
+		return err
+	}
+	hasChanged = hasChanged || changed
+
+	// SSL enabled. Needed to set specifically the ssl port setting starting with Nautilus.
+	if ssl == "true" {
+		changed, err = client.MgrSetConfig(c.context, c.Namespace, m.DaemonID, c.clusterInfo.CephVersion, "mgr/dashboard/ssl_server_port", port, false)
+		if err != nil {
+			logger.Infof("Error trying to set Ceph Dashboard SSL port value. The Ceph dashboard module default value for this setting will be used. Error details: %+v", err)
```

Comment:
`logger.Warningf`?

Reply:
I have removed the messages because now the setting is only read if we are sure that is going to be present (ceph version >= 14.2.1)